numpy/core/src/umath/loops_arithmetic.dispatch.c.src

```diff
@@ -0,0 +1,132 @@
+/*@targets
+ ** $maxopt baseline
+ ** sse2 sse41 avx2 avx512_skx
+ ** vsx2
+ ** neon
+ **/
+#define _UMATHMODULE
+#define _MULTIARRAYMODULE
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+
+#include "simd/simd.h"
+#include "loops_utils.h"
+#include "loops.h"
+#include "lowlevel_strided_loops.h"
+#include<signal.h> 
+// Provides the various *_LOOP macros
+#include "fast_loop_macros.h"
+
+//###############################################################################
+//## Unsigned Integers
+//###############################################################################
+/********************************************************************************
+ ** Defining the SIMD kernels
+ ********************************************************************************/
+#ifdef NPY_SIMD
+/**begin repeat
+ *  #sfx = u8, u16, u32, u64#
+ *  #type = npy_ubyte, npy_ushort, npy_uint, npy_ulong#
+ */
+
+static void simd_divide_by_scalar_contig_contig_@sfx@
+(const char *src, const @type@ scalar, char *dst,
+ int len)
+{
+    npyv_lanetype_@sfx@ *npyv_src = (npyv_lanetype_@sfx@ *) src;
+    npyv_lanetype_@sfx@ npyv_scalar = (npyv_lanetype_@sfx@) scalar;
+    npyv_lanetype_@sfx@ *npyv_dst = (npyv_lanetype_@sfx@ *) dst;
+    const int vstep = npyv_nlanes_@sfx@;
+    const npyv_@sfx@x3 divisor = npyv_divisor_@sfx@(npyv_scalar);
+
+    for (; len >= vstep; len -= vstep, npyv_src += vstep, npyv_dst += vstep) {
+        npyv_@sfx@ a = npyv_load_@sfx@(npyv_src);
+        npyv_@sfx@ c = npyv_divc_@sfx@(a, divisor);
+        npyv_store_@sfx@(npyv_dst, c);
+    }
+    for (; len > 0; --len, ++npyv_src, ++npyv_dst) {
+        const npyv_lanetype_@sfx@ a = *npyv_src;
+        *npyv_dst = a / npyv_scalar;
+    }
+    npyv_cleanup();
+}
+
+/**end repeat**/
+#endif
+
+/**begin repeat
+ * Unsigned types
+ *  #type = npy_ubyte, npy_ushort, npy_uint, npy_ulong, npy_ulonglong#
+ *  #TYPE = UBYTE, USHORT, UINT, ULONG, ULONGLONG#
+ *  #SIGNED_TYPE = BYTE, SHORT, INT, LONG, LONGLONG#
+ */
+#if NPY_BITSOF_@SIGNED_TYPE@ <= 8
+    #define simd_divide_by_scalar_@type@ simd_divide_by_scalar_contig_contig_u8
+#elif NPY_BITSOF_@SIGNED_TYPE@ <= 16
+    #define simd_divide_by_scalar_@type@ simd_divide_by_scalar_contig_contig_u16
+#elif NPY_BITSOF_@SIGNED_TYPE@ <= 32
+    #define simd_divide_by_scalar_@type@ simd_divide_by_scalar_contig_contig_u32
+#else
+    #define simd_divide_by_scalar_@type@ simd_divide_by_scalar_contig_contig_u64
+#endif
+static NPY_INLINE int
+run_binary_simd_divide_@TYPE@(char **args, npy_intp const *dimensions, npy_intp const *steps)
+{
+    BINARY_DEFS
+
+    if (n == 0) {
+        return 1;
+    }
+
+    const @type@ in2 = *(@type@ *)ip2;
+    if (in2 == 0) {
+        npy_set_floatstatus_divbyzero();
+        BINARY_LOOP_SLIDING {
+            *((@type@ *)op1) = 0;
+        }
```

Comment:
What I understand here is if the "first" divisor is zero then fill all output with zeros.

You should move it to the branch `IS_BLOCKABLE_BINARY_SCALAR2` or just escape it to be handled by the main binary loop its a strange behavior anyway and we shouldn't optimize it.

I guess that what makes `masked_array` test fails.